virl2_client/models/auth_management.py

```diff
@@ -0,0 +1,391 @@
+
+    def _get_current_settings(self) -> dict:
+        settings = self._session.get(self._CONFIG_URL).json()
+        del settings["method"]
+        return settings
+
+    def get_setting(self, setting: str) -> Any:
+        """
+        Returns the value of a specific setting of this authentication method.
+        Note: consider using parameters instead.
+        """
+        self.sync_if_outdated()
+        return self._settings[setting]
+
+    def get_settings(self) -> dict:
+        """Returns a dictionary of the settings of this authentication method."""
+        self.sync_if_outdated()
+        return self._settings.copy()
+
+    def _update_setting(self, setting: str, value) -> None:
+        """
+        Update a setting for this auth method.
+
+        :param setting: the setting to update
+        :param value: the value to set the setting to
+        """
+        if setting == "method":
+            raise ValueError(
+                f"To set method, use 'AuthManagement.method = {value}' instead."
+            )
+        # TODO: currently, method is required, so we have to provide it;
+        #  do we provide this one or the one that is currently active?
+        #  in other words, if method=local and you change an LDAP setting, do we
+        #  switch to LDAP or keep local?
+        #  we can also provide an argument, so the user can choose, but what would
+        #  the default be?
+        # also note this should generally not happen;
+        # see AuthManagement.method above
+        method = self._METHOD
+        self._session.put(
+            self._CONFIG_URL,
+            json={setting: value, "method": method},
+        )
+        self._auth_management._method = AuthMethod(method)
+        if setting in self._settings:
+            self._settings[setting] = value
+
+    def update_settings(self, settings_dict: Optional[dict] = None, **kwargs) -> None:
+        """
+        Update multiple auth settings at once.
+        More efficient in batches than setting manager properties.
+        If passed a dictionary, reads the dictionary for settings.
+        If passed multiple keyword arguments, each is taken as a setting.
+        Can mix and match; keywords take precedence.
+
+
+        Examples::
+            am = client.auth_management
+            am.update_settings({"method": "ldap", "verify_tls": False})
+            am.update_settings(method="ldap", verify_tls=False)
+            # "verify_tls" key in dictionary overridden by keyword argument into False
+            am.update_settings({"method": "ldap", "verify_tls": True}, verify_tls=False)
+
+        :param settings_dict: a dictionary of settings
+        :param kwargs: keywords of settings
+        """
+        settings = {}
+        if settings_dict:
+            settings.update(settings_dict)
+        settings.update(kwargs)
+        if not settings:
+            raise TypeError("No settings to update.")
+        if settings.get("method", None):
```

Comment:
None is the default value.

Reply:
Right, pop raises by default, not get, thanks.